### PR TITLE
[SPARK-16241] [ML] model loading backward compatibility for ml NaiveBayes

### PR DESCRIPTION
**Pull request by zlpmichelle**
## What changes were proposed in this pull request?

model loading backward compatibility for ml NaiveBayes
## How was this patch tested?

existing ut and manual test for loading models saved by Spark 1.6.

### CODE REVIEW

**Comment by hhbyyh** [comment]
LGTM.

**Comment by zlpmichelle** [comment]
Thanks @hhbyyh @AmplabJenkins!

**Comment by yanboliang** [comment]
Jenkins, test this please.

**Comment by mengxr** [comment]
ok to test

File: mllib/src/main/scala/org/apache/spark/ml/classification/NaiveBayes.scala [hunk 1]
```diff
@@ -28,8 +28,9 @@ import org.apache.spark.ml.util._
 import org.apache.spark.mllib.classification.{NaiveBayes => OldNaiveBayes}
 import org.apache.spark.mllib.classification.{NaiveBayesModel => OldNaiveBayesModel}
 import org.apache.spark.mllib.regression.{LabeledPoint => OldLabeledPoint}
+import org.apache.spark.mllib.util.MLUtils
 import org.apache.spark.rdd.RDD
-import org.apache.spark.sql.Dataset
+import org.apache.spark.sql.{Row, Dataset}
```
Comment (yanboliang) [inline mllib/src/main/scala/org/apache/spark/ml/classification/NaiveBayes.scala]: Notice the import order, it should be ```import org.apache.spark.sql.{Dataset, Row}```

**Comment by zlpmichelle** [comment]
Fixed it. Thanks @yanboliang for your nice review.

**Comment by zlpmichelle** [comment]
The patch passed all tests. Thanks very much @mengxr !

**Comment by yanboliang** [comment]
LGTM, merged into master and branch-2.0. Thanks!

**Comment by lipingzh** [comment]
Thanks Yanbo.

**Pull request closed without merging**